### ISSUE TITLE
4 Teams / 3 Teams multiple problems

### ISSUE THREAD

**Issue by lemonyPRIDE** [efroemling/ballistica#207]
Hello Eric,
Hope you're safe and sound. So I wanted to post this issue a long time for now as this issue is related to one of the ideas that a lot of players actually enjoy in the PUBLIC. So I'll state the issues for making of 3 Teams and 4 Teams server. The version we're running the servers are 1.4 yet, as 1.5 is still in its early stage. 

1) **Lives Balancing -** I noticed that whenever multiple teams play elimination game, lives are not balanced across all the teams except for the teams named as team1 and team2 in the script, but team3 and team4 gets ruled out. So I need your help here.

2) **Unstable Spawn for Solo -** The players would spawn at normal specified spawn area first, but after one of them dies, the next upcoming player would spawn randomly on one of the spawn area of the specified team, even if one player is already present at that location. So, I tried removing the spawn randomness loop and works but its temporary. Refer to the screenshot below.

![InkedWhatsApp Image 2020-11-07 at 12 38 09_LI](https://user-images.githubusercontent.com/67043009/98434686-00a93700-20f8-11eb-8f1d-f09c6e7b8062.png)
 

3) **UI** - Player icon overlap each other whether its solo or its elimination. Also need help on this one. For better understanding, refer to the screenshots I shared both above and below.

![BombSquad 11_7_2020 12_30_24 PM_LI](https://user-images.githubusercontent.com/67043009/98434448-b1fa9d80-20f5-11eb-88b4-99a9f3223c90.jpg)

I know you're working really hard on the 1.5 and upcoming 1.6 update, but if you help put with these issue here, others can simply re-direct to this issue and work there way out.

**Comment by DrovGamedev** [comment]
**Solve** it by yourself🤪

**Comment by efroemling** [comment]
To clarify: all of these issues apply only when there's more than 2 teams?
With the 1.5 update I've tried to move closer to a place where multi-team games are officially supported. (you can see in the [docs](https://github.com/efroemling/ballistica/blob/master/docs/ba_module.md) that there's now a ba.MultiTeamSession class which both 2-team and ffa inherit from, etc.).

I imagine each of those games just needs to be updated slightly to do the right thing when there's more than 2 teams present. If anyone would like to help with that work here I'd be happy to review/accept it. Or even just demonstrating which of those same issues still exist in 1.5 would be helpful. I can try to address some of them myself, but I must admit its not very high priority compared to other things I'm working on right now. That also unfortunately applies to 1.4; I'd be spreading myself too thin trying to do fixes or improvements on that codebase at this point.

Regardless, thanks for bringing this to my attention and hopefully we can get this stuff behaving better for multi-team at some point..

**Comment by lemonyPRIDE** [comment]
Yes indeed, these problems arise only when there's more than one team. 
I'll try this on 1.5 and report it as sson as possible. Also, does that mean if I tweak the ba.MultiTeamSession, all these problem would be solved?

**Comment by lemonyPRIDE** [comment]
@efroemling I found out that multiple team spawning and everything is still same as the bs1.4, every map has to be updated with new respawn points and everything yet, all the problems are still the same. I fixed the solo spawn problem by removing the spawn loop. But for UI and unbalanced lives problem still persists. Am sure if the solo spawn was this simple, others wont be a problem, so can you assist in this problem by pointing to the files where I can edit the UI which updates the lives on the  screen?

**Comment by Moonard1** [comment]
How to get bombsquad teams script working

**Comment by MatiasMGz** [comment]
Any updates on this?
